plugins/org.eclipse.n4js.ts/src/org/eclipse/n4js/ts/utils/TypeHelper.xtend

```diff
@@ -194,4 +200,46 @@ public class TypeHelper {
 		return -1;
 	}
 
+
+	/**
+	 * Extracts the type(s) referenced by a type ref.
+	 * @param typeRef
+	 *  		the type ref, from which the type should be extracted.
+	 * @return
+	 * 			the list of extracted types. There can be multiple returned types due to composed type refs.
+	 */
+	public def List<Type> extractType(TypeArgument typeArg) {
```

Comment:
1. please double-check that we do not already have something like this, somewhere.
2. rename to `extractTypes` (or maybe `collectTypes`).
3. this does not seem to require injection, so make static and move to `TypeUtils`.